src/Nethermind/Nethermind.Core/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nethermind.Core.Extensions
+{
+    public static class TypeExtensions
+    {
+       public static Type GetDirectInterfaceImplementation(this Type interfaceType)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new NotSupportedException("This method is only allowed to use on interface types");
```

Comment:
Great practice would be to display the actual type received so if error occurs it can be identified quickly.